File: backend-api/src/main/java/com/smartenergy/backendapi/model/RegistroOperativo.java
package com.smartenergy.backendapi.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Getter
@Setter
@Entity
@Table(name = "RegistroOperativo")
public class RegistroOperativo extends BaseEntity {

    @Enumerated(EnumType.STRING)
    private Estado estado;

    private double consumo;
    private LocalDateTime inicio;
    private LocalDateTime fin;

    @ManyToOne
    @JoinColumn(name = "equipo_id")
    private Equipo equipo;
}

File: mobile-app/app/src/main/java/com/example/smartenergy/model/RegistroOperativo.kt
package com.example.smartenergy.model

import java.time.LocalDateTime
import java.util.UUID

data class RegistroOperativo(
    val id: String? = null,
    val estado: Estado,
    val consumo: Double,
    val inicio: LocalDateTime,
    val fin: LocalDateTime? = null,
    val equipo: Equipo? = null
)
